src/main/java/htsjdk/samtools/SAMTextHeaderCodec.java

```diff
@@ -437,8 +441,8 @@ private void writeHDLine(final boolean keepExistingVersionNumber) {
         println(StringUtil.join(FIELD_SEPARATOR, fields));
     }
 
-    private void writeSQLine(final SAMSequenceRecord sequenceRecord) {
-        final int numAttributes =sequenceRecord.getAttributes() != null ? sequenceRecord.getAttributes().size() : 0;
+    void writeSQLine(final SAMSequenceRecord sequenceRecord) {
```

Comment:
hmmm. I'm not so happy about the access change of this internal function. could you perhaps wrap it with a public function that also checks that the writer is not null and is more cognizant of it's package-protected status? For example, a java-doc would be good that would inform the assumptions in place (like the existence of a writer..)